### PR TITLE
:bug: Source Zendesk Support: Fix long-running syncs for Ticket Metrics, Ticket Audits and Satisfaction Ratings streams.

### PR DESCRIPTION
**Pull request by darynaishchenko**
oncall: https://github.com/airbytehq/oncall/issues/4740

Sync stuck due to slow reading and filtering in streams.

**Stream Satisfaction ratings:** 
Zendesk API returns next and previous urls with `start_time` param, so we don't need to remove it for next pages.
Example:
```
"links": {
        "prev": "https://domain.zendesk.com/api/v2/satisfaction_ratings.json?page%5Bbefore%5D=yyy%5Bsize%5D=100&start_time=1711095890",
        "next": "https://domain.zendesk.com/api/v2/satisfaction_ratings.json?page%5Bafter%5D=xxx&page%5Bsize%5D=100&start_time=1711095890"
    }
```
**Stream Ticket Metrics:**
Changed to use parent stream Tickets in slices to retrieve metrics and audits by specific ticket. Slices are retrieved with date filter so  only info by recently updated tickets will be synced. This change decreases sync time in case of a lot of records as source doesn't need to read all available data and filter it. 

**Stream Ticket Audits:** 
Fixed with new `_validate_response` method in `_read_pages` to stop reading pages when response from API contains records older than start date or cursor value.

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]
lgtm

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py [hunk 1]
```diff
@@ -539,8 +539,13 @@ def request_params(
...
     ) -> MutableMapping[str, Any]:
-        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
-        params.update({"sort_by": "asc"})
+        params = {
+            "start_time": self.get_stream_state_value(stream_state),
+            "page[size]": self.page_size,
+        }
+        if next_page_token:
+            params.update(next_page_token)
+        params.update({"sort_by": "created_at"})
```
Comment (tolik0) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:548]:
```suggestion
        params = {
            "start_time": self.get_stream_state_value(stream_state),
            "page[size]": self.page_size,
            "sort_by": "created_at",
        }
        if next_page_token:
            params.update(next_page_token)
```

**Review by tolik0 — APPROVED** [review]

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py [hunk 2]
```diff
@@ -643,6 +648,41 @@ def read_records(
...
+    def _validate_response(self, response: requests.Response, stream_state: Mapping[str, Any]) -> bool:
+        """
+        Ticket Audits endpoint doesn't allow filtering by date, but all data sorted by descending.
+        This method used to stop making requests once we receive a response with cursor value greater than actual cursor.
+        This action decreases sync time as we don't filter extra records in parse response.
+        """
+        data = response.json().get(self.response_list_name)
+        created_at = data[0].get(self.cursor_field)
+        cursor_date = (stream_state or {}).get(self.cursor_field) or self._start_date
+        return created_at > cursor_date
```
Comment (tolik0) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:660]: Should it be `>=` to avoid missing any records?

Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:660]: good point, will add it

**Review by tolik0 — APPROVED** [review]

**Review by askarpets — APPROVED** [review]

**Pull request merged**